### PR TITLE
Individual keyword chips in search bar for each new keyword search

### LINKED ISSUE #349: individual facet chips for new keyword searches

### PR DESCRIPTION
**Pull request by jfranco-gri**
Closes #349 

**Reviewers**
- [x] @joshuago78 
- [x] @adamcee

### CODE REVIEW

File: src/client/app/search/search.controller.js [hunk 1]
```diff
@@ -144,7 +144,9 @@
       var newQuery;
       if (query) {
         query = query.trim();
-        $scope.queryTerms.push(query);
+        if ($scope.queryTerms.indexOf(query) == -1){
```
Comment (adamcee) [inline src/client/app/search/search.controller.js]: Could you please change `==` to `===`? 

Also, this won't cover if a URL w/a dupe q param is put in the address bar like `/search?q=art&q=art`. I know that is an extreme edge case but it will still break things ...

File: src/client/app/advanced_search/advanced-search.controller.js [hunk 2]
```diff
@@ -67,19 +67,25 @@
...
+          advFields.push(
             searchService.buildAdvancedField(filter.field, filter.term)
           );
         }
       });
 
+      var opts = {
+        q: query,
+        advancedFields: advFields
+      }
```
Comment (adamcee) [inline src/client/app/advanced_search/advanced-search.controller.js]: Please add semi-colon here at end of var declaration ; )

File: src/client/app/search/search.controller.js [hunk 3]
```diff
@@ -143,25 +143,16 @@
     $scope.newQuerySearch = function(query){
       var newQuery;
       if (query) {
-        newQuery = query.trim();
-        if ($scope.queryTerm) {
-          newQuery = $scope.queryTerm + " " + newQuery;
+        query = query.trim();
+        if ($scope.queryTerms.indexOf(query) == -1){
```
Comment (adamcee) [inline src/client/app/search/search.controller.js]: Could you please change `== `to `===` ?

Also, this won't cover if a URL w/a dupe q param is put in the address bar like /search?q=art&q=art. I know that is an extreme edge case but it will still break the search box ... maybe move this code to the resolve or something?

Lastly could you add a unit test to check that dupe query terms are removed as a check against regression?

**Comment by jfranco-gri** [comment]
Okay, fixed the issue with duplicate queries and added unit and e2e tests.

File: src/client/app/app.config.routing.js [hunk 4]
```diff
@@ -48,13 +49,25 @@
...
 
-            opts.q = $stateParams.q;
             opts.size = parseInt($stateParams.size);
             opts.from = parseInt($stateParams.from);
             opts.sort = $stateParams.sort;
             opts.date.gte = $stateParams.date_gte;
             opts.date.lte = $stateParams.date_lte;
 
+            opts.q = [];
+            if ($stateParams.q) {
```
Comment (adamcee) [inline src/client/app/app.config.routing.js]: nice

File: src/client/app/app.config.routing.js [hunk 5]
```diff
@@ -48,13 +49,25 @@
...
+              var seen = {};
+              opts.q = $stateParams.q.filter(function(query){
+                if (seen[query]) {
+                  return false;
+                } else {
+                  seen[query] = true;
+                  return true;
+                }
```
Comment (adamcee) [inline src/client/app/app.config.routing.js]: nice

**Comment by adamcee** [comment]
Looks good 😸
